cogniac/edgeflow.py

```diff
@@ -259,7 +259,16 @@ def upgrade(self, software_version):
         args = dict()
         if software_version is not None:
             args['software_version'] = software_version
-        self._cc._post("/gateways/%s/event/upgrade" % (self.gateway_id), data=args)
+        self._cc._post("/gateways/%s/event/upgrade" % (self.gateway_id), json=args)
+
+    def rollback(self, software_version):
```

Comment:
how about we mirror the endpoint name w/ set_boot_software_version()?   this can be used to roll back or forward as long as the software is downloaded.